instrumentation/grape/lib/opentelemetry/instrumentation/grape/instrumentation.rb

```diff
@@ -21,18 +22,10 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
         end
 
         compatible do
-          # ActiveSupport::Notifications were introduced in Grape 0.13.0
-          # https://github.com/ruby-grape/grape/blob/master/CHANGELOG.md#0130-2015810
           gem_version >= MINIMUM_VERSION
         end
 
-        # Config options available in ddog Grape instrumentation
-        option :enabled, default: true, validate: :boolean
-        option :error_statuses, default: [], validate: :array
```

Comment:
This is an option provided by the DDog instrumentation (more info [here](https://docs.datadoghq.com/tracing/trace_collection/dd_libraries/ruby/#grape)). I've looked into this but I don't think it adds a lot of value in our case now. It would add a lot of code and complexity, and we can always come back to it if we identify the need.

Basically, DDog is using a [custom matcher](https://github.com/DataDog/dd-trace-rb/blob/117df995b1d624db4844db87ac7b547e1d65f6a1/lib/datadog/tracing/contrib/status_code_matcher.rb#L12)  that we would need to reproduce to be able to support this option. The only benefit would be that we could specify which status codes mean that the [span status](https://opentelemetry.io/docs/concepts/signals/traces/#span-status) is set to error.
IMO, this does not add much. With the implementation proposed here, span status is set to `error` only when an error is raised. In the meantime, they are `unset` (the default value). Assuming that the error is caught somewhere in the trace, then the end status would be caught there. And also by setting `http.status_code` as an attribute when no error is raised (something that is not present in the DDog implementation), we have another way of identifying if the request was not successful.